### PR TITLE
Handle IPv6 route types such as anycast, multicast when returned from IPv6 route table queries

### LINKED ISSUE #60232: [BUG] network.default_route: 'anycast' does not appear to be an IPv4 or IPv6 network
**Description**
I have got an error (**ValueError: 'anycast' does not appear to be an IPv4 or IPv6 network**) when executing network.default_route:

<details><summary><code>salt minid network.default_route</code></summary>

```
    The minion function caused an exception: Traceback (most recent call last):
      File "/usr/lib/python3/dist-packages/salt/minion.py", line 1905, in _thread_return
        function_name, function_args, executors, opts, data
      File "/usr/lib/python3/dist-packages/salt/minion.py", line 1861, in _execute_job_function
        return_data = self.executors[fname](opts, data, func, args, kwargs)
      File "/usr/lib/python3/dist-packages/salt/loader.py", line 1235, in __call__
        return self.loader.run(run_func, *args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/loader.py", line 2268, in run
        return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
      File "/usr/lib/python3/dist-packages/contextvars/__init__.py", line 38, in run
        return callable(*args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/loader.py", line 2283, in _run_as
        return _func_or_method(*args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/executors/direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/loader.py", line 1235, in __call__
        return self.loader.run(run_func, *args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/loader.py", line 2268, in run
        return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
      File "/usr/lib/python3/dist-packages/contextvars/__init__.py", line 38, in run
        return callable(*args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/loader.py", line 2283, in _run_as
        return _func_or_method(*args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1807, in default_route
        _routes = routes()
      File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1764, in routes
        routes_ = _ip_route_linux()
      File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 609, in _ip_route_linux
        address_mask = convert_cidr(comps[0])
      File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1232, in convert_cidr
        cidr = calc_net(cidr)
      File "/usr/lib/python3/dist-packages/salt/modules/network.py", line 1255, in calc_net
        return __utils__["network.calc_net"](ip_addr, netmask)
      File "/usr/lib/python3/dist-packages/salt/loader.py", line 1235, in __call__
        return self.loader.run(run_func, *args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/loader.py", line 2268, in run
        return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
      File "/usr/lib/python3/dist-packages/contextvars/__init__.py", line 38, in run
        return callable(*args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/loader.py", line 2283, in _run_as
        return _func_or_method(*args, **kwargs)
      File "/usr/lib/python3/dist-packages/salt/utils/network.py", line 1185, in calc_net
        return str(ipaddress.ip_network(ipaddr, strict=False))
      File "/usr/lib/python3/dist-packages/salt/ext/ipaddress.py", line 108, in ip_network
        address)
    ValueError: 'anycast' does not appear to be an IPv4 or IPv6 network
```

</details>

**Setup**

**Steps to Reproduce the behavior**

**Expected behavior**
Should not throw

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
Salt Version:
          Salt: 3003
 
Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: 2.7.3
     docker-py: 4.4.1
         gitdb: 4.0.5
     gitpython: 3.1.13
        Jinja2: 2.10.1
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.6.1
        pygit2: Not Installed
        Python: 3.8.5 (default, Jan 27 2021, 15:41:15)
  python-gnupg: 0.4.5
        PyYAML: 5.3.1
         PyZMQ: 18.1.1
         smmap: 3.0.5
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.2
 
System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.0-73-generic
        system: Linux
       version: Ubuntu 20.04 focal

```
</details>

**Additional context**

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Allows for anycast, mulitcast etc to be handled when returned from IPv6 route tables, for example:
```
ip -6 route show table all
```
in default_route.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/60232

### Previous Behavior
Encountering anycast or multicast would throw errors since they we not ignored.

### New Behavior
IPv6 route types for anycast, multicast and others are now ignored and no longer throw errors

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by garethgreenaway — DISMISSED** [review]

File: salt/modules/network.py
Comment (waynew) [inline salt/modules/network.py:599]: **suggestion**

If comps[0] in ('anycast', 'multicast', ....):
    continue

Would that clean things up a bit?

Comment (dmurphy18) [reply salt/modules/network.py:585]: done

**Review by bryceml — COMMENTED** [review]
lgtm, I'm not very familiar with this code so we should probably get at least one more approval.

**Review by bryceml — DISMISSED** [review]

**Review by Ch3LL — CHANGES_REQUESTED** [review]
this needs a changelog

**Comment by dmurphy18** [comment]
@Ch3LL Added changelog/60232.fixed

File: salt/modules/network.py
Comment (waynew) [reply salt/modules/network.py:585]: :rocket: nice!

Comment (waynew) [reply salt/modules/network.py:585]: (and, that's why we review each others' code, to help catch those things :slightly_smiling_face: )

**Review by waynew — DISMISSED** [review]
:rocket:

File: salt/modules/network.py
Comment (Ch3LL) [inline salt/modules/network.py:1835]: These variables are unused.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/modules/network.py
Comment (dmurphy18) [reply salt/modules/network.py:1835]: done, sorry missed left over debug line

File: salt/modules/network.py
Comment (Ch3LL) [inline salt/modules/network.py:1835]: This variable is unused as well.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/modules/network.py
Comment (dmurphy18) [reply salt/modules/network.py]: Gone through that area and ensured there should be no more droppings left from debugging the issue.
Sorry for that.

**Review by Ch3LL — APPROVED** [review]

**Comment by Ch3LL** [comment]
ping @waynew would you mind re-reviewing here before merge

**Review by waynew — APPROVED** [review]

**Pull request merged**
